modules/host/announce.go

```diff
@@ -16,22 +16,27 @@ var (
 	errUnknownAddress = errors.New("host cannot announce, does not seem to have a valid address.")
 )
 
-// announce creates an announcement transaction and submits it to the network.
-func (h *Host) announce(addr modules.NetAddress) error {
+// managedAnnounce creates an announcement transaction and submits it to the network.
+func (h *Host) managedAnnounce(addr modules.NetAddress) error {
```

Comment:
This should be in another PR, but I had finished coding it before I realized that the problem was in more places than just this.

Basically it's a violation of conventions to call `AcceptTransactionSet` while the host is under lock (locking an external package while retaining a lock over your own package is not allowed) and in this case the violation caused a deadlock.

Unfortunately, the host contract negotiation stuff has this problem in a much more deeply embedded way =/. Wallet also appears to have this problem, and in each case they cause deadlocks.